src/DBmysql.php

```diff
@@ -1543,16 +1543,29 @@ public function updateOrDie($table, $params, $where, $message = '', array $joins
      * @return mysqli_result|boolean Query result handler
      */
     public function updateOrInsert($table, $params, $where, $onlyone = true)
+    {
+        $req = $this->request(array_merge(['FROM' => $table], $where));
+        $data = array_merge($where, $params);
+
+        try {
+            $query = $this->buildUpdateOrInsert($table, $data, $where, $onlyone);
+            return $this->doQueryOrDie($query, 'Unable to create new element or update existing one');
+        } catch (\RuntimeException $e) {
+            trigger_error($e->getMessage(), E_USER_WARNING);
+            return false;
+        }
+    }
+
+    public function buildUpdateOrInsert($table, $params, $where, $onlyone = true): string
```

Comment:
I ended up not needing this new method, I've still kept it as it may be useful to someone else one day.